tarkov/inventory/inventory.py

```diff
@@ -349,74 +360,88 @@ def _is_item_in_root(self, item: Item) -> bool:
         """
         return isinstance(item.location, ItemInventoryLocation) and item.parent_id == self.inventory.root_id
 
-    def remove(self, item: Item, child_items: List[Item]) -> None:
+    def get(self, x: int, y: int) -> bool:
+        if not 0 <= x < self.width:
+            raise IndexError
+        if not 0 <= y < self.height:
+            raise IndexError
+
+        return self.map[x][y]
+
+    def set(self, x: int, y: int, state: bool) -> None:
+        if self.map[x][y] == state:
+            raise self.InvalidCellStateError(f"Cell {x} {y} already has {state} state")
+        self.map[x][y] = state
+
+    def _get_parent_item_in_inventory_root(self, item: Item) -> Item:
         parent_item = item
-        while not self._is_item_in_root(parent_item) and parent_item.parent_id is not None:
+        while parent_item.parent_id != self.inventory.root_id and parent_item.parent_id is not None:
             parent_item = self.inventory.get(parent_item.parent_id)
+        return parent_item
 
-        # Recalculate parent item footprint, needed for disassembling
-        try:
-            del self.footprints[parent_item.id]
-        except KeyError:
-            pass
+    def remove(self, item: Item, child_items: List[Item]) -> None:
```

Comment:
I don't have another suggestion appart from better commenting this method and going back to it at a later time to optimize it when things are more "definitive".